manager/controllers/app/data_catalog_interface.go

```diff
@@ -81,10 +81,11 @@ func (r *M4DApplicationReconciler) RegisterAsset(catalogID string, info *app.Dat
 	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 	defer cancel()
 
-	datasetDetails := &pb.DatasetDetails{}
-	if err := serde.FromRawExtention(info.Details, datasetDetails); err != nil {
-		return "", err
+	datasetDetails, ok := info.Details.Data.(*pb.DatasetDetails)
```

Comment:
@shlomitk1 this would probably not work, I will fix it in a separate PR. Do we have a test that uses register asset?

Reply:
We don't.